crates/ruff_python_parser/src/lexer.rs

```diff
@@ -181,16 +179,27 @@ impl<'source> Lexer<'source> {
                 return Ok(self.lex_fstring_start(quote, true));
             }
             (_, quote @ ('\'' | '"')) => {
-                if let Ok(string_kind) = StringKind::try_from(first) {
+                if let Ok(string_kind) = StringFlags::try_from(first) {
                     self.cursor.bump();
                     return self.lex_string(string_kind, quote);
                 }
             }
             (_, second @ ('r' | 'R' | 'b' | 'B')) if is_quote(self.cursor.second()) => {
                 self.cursor.bump();
-                if let Ok(string_kind) = StringKind::try_from([first, second]) {
+                let flags = StringFlags::try_from(first);
+                let flags = {
+                    if second == 'r' || second == 'R' {
+                        flags.map(StringFlags::with_r_prefix)
+                    } else {
+                        flags.map(StringFlags::with_b_prefix)
+                    }
+                };
+                if let Ok(Ok(mut flags)) = flags {
                     let quote = self.cursor.bump().unwrap();
-                    return self.lex_string(string_kind, quote);
+                    if quote == '"' {
+                        flags = flags.with_double_quotes();
```

Comment:
Should this happen in `lex_string`, instead? `lex_string` currently sets the triple-quoting, for example.